src/components/program/ProgramDataBarDetails.jsx

```diff
@@ -0,0 +1,70 @@
+import React, {
+  useContext, useRef, useState, useEffect,
+} from 'react';
+import PropTypes from 'prop-types';
+import { ProgramContext } from './ProgramContextProvider';
+import { appendProgramToProgramType } from './data/utils';
+
+const ProgramDataBarDetails = ({ handleStick, handleRelease }) => {
+  const {
+    program: {
+      title, authoringOrganizations: owners, type,
+    },
+  } = useContext(ProgramContext);
+
+  const wrapper = useRef(null);
+  const [sticky, setSticky] = useState(false);
+  const [componentTop, setComponentTop] = useState(0);
+
+  const getOffsetTop = () => {
+    if (wrapper?.current) {
+      const rect = wrapper.current.getBoundingClientRect();
+      const scrollTop = window.pageYOffset || document.documentElement.scrollTop;
+
+      return rect.top + scrollTop;
+    }
+
+    return 1;
+  };
+
+  const handleScroll = () => {
+    const windowPosition = window.scrollY;
+    const wrapperTop = getOffsetTop();
+    if (!sticky) {
+      if (windowPosition >= wrapperTop) {
+        setSticky(() => true);
+        setComponentTop(() => wrapperTop);
+        handleStick();
+      }
+    }
+    if (windowPosition === 0 || (windowPosition + 30) < componentTop) {
+      setSticky(() => false);
+      handleRelease();
+    }
+  };
```

Comment:
Can you explain what's happening here? Maybe add a comment too

Reply:
added comments